### ISSUE TITLE
Modify Gateway CopyObject API

### ISSUE THREAD

**Issue by N-o-Z** [treeverse/lakeFS#4478]
Prerequisite: [Implement CopyObject API](https://github.com/treeverse/lakeFS/issues/4477#top)

For copy operations on different branches, change CopyObject behavior to perform "full copy" using the underlying storage adapter's Copy method.

1. Copy object in the same branch will work the same - creating a new staging entry using the existing entry information.
2. For objects that are not part of the branch, use the underlying adapter copy operation.
